pkg/scheduler/algorithm/priorities/selector_spreading.go

```diff
@@ -250,43 +235,60 @@ func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityMap(pod *v1.Pod, meta
 // CalculateAntiAffinityPriorityReduce computes each node score with the same value for a particular label.
 // The label to be considered is provided to the struct (ServiceAntiAffinity).
 func (s *ServiceAntiAffinity) CalculateAntiAffinityPriorityReduce(pod *v1.Pod, meta interface{}, sharedLister schedulerlisters.SharedLister, result framework.NodeScoreList) error {
+	reduceResult := make([]float64, len(result))
```

Comment:
We need good documentation explaining why we have this awkward implementation: that we are trying to workaround a v1 API that allows defining only a single label, but multiple custom priorities etc.

Reply:
Added something, let me know if it look good to you.